### ISSUE TITLE
Warn if $GOOGLE_APPLICATION_CREDENTIALS is set

### ISSUE THREAD

**Issue by eriknolte** [terraform-google-modules/terraform-example-foundation#925]
### TL;DR

GCP access tokens can come from two places: (1) the file pointed to by $GOOGLE_APPLICATION_CREDENTIALS or (2) the output of `gcloud auth application-default print-access-token`.  If set, the env var token is preferred to the gcloud auth token.  The validate-requirements.sh script expects access tokens to come from (2).

I had a lower privileged access token used by other projects set in $GOOGLE_APPLICATION_CREDENTIALS which overrode my Organization Admin privileges in IAM.  This caused validate-requirements.sh to finish successfully, but `terraform apply` to fail with a dozen resource permission errors.

### Detailed design

```markdown
Consider adding something like this in validate-requirements.sh's  validate_gcloud_configuration() function:

if [ -n "$GOOGLE_APPLICATION_CREDENTIALS" ]
then
    echo "  warning: using \$GOOGLE_APPLICATION_CREDENTIALS for access token."
    echo "  If you encounter permission issues either unset the env var or update the token."
    echo "  See https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login"
fi
```
